### PR TITLE
Fix #9595

### LINKED ISSUE #9595: [ESIL] Calculated address is 64 bits instead of 32 bits when overflowed
| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | macOS 10.13.3
| File format of the file you reverse (mandatory)      | PE
| Architecture/bits of the file (mandatory)            | x86/32
| r2 -v full output, **not truncated** (mandatory)     | radare2 2.4.0-git 17460 @ darwin-x86-64 git.2.2.0-642-g348057f5a
commit: 348057f5ae22c0694b83fd51bc16760d51b1553c build: 2018-03-02__07:38:26

### Expected behavior
When trying to access an address in memory, radare2 should take into account the maximum bit size of the targeted architecture.

So for example:

```bash
mov byte [esi + eax], dl
```
with:
```bash
eax 0x00419aa8
esi 0xffd84980
```

```esi+eax```should give 0x0019e428

### Actual behavior
When doing this calculation, radare2 is accessing 0x10019e428.

### PR DESCRIPTION
**Pull request by redmed666**
Assure that the address accessed when writing or reading is logical with asm.bits for the analyzed file.

### CODE REVIEW

**Comment by condret** [comment]
this breaks gameboy

File: libr/anal/esil.c [hunk 1]
```diff
@@ -356,6 +361,7 @@ static int internal_esil_mem_write(RAnalEsil *esil, ut64 addr, const ut8 *buf, i
 
 static int internal_esil_mem_write_no_null(RAnalEsil *esil, ut64 addr, const ut8 *buf, int len) {
 	int ret = 0;
+	addr = addr & esil->addrmask;
```
Comment (radare) [inline libr/anal/esil.c]: use &= instead of addr = addr & ..

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -1257,6 +1257,8 @@ static void core_anal_bytes(RCore *core, const ut8 *buf, int len, int nops, int
 	RAnalOp op;
 	ut64 addr;
 	bool isFirst = true;
+	int addrsize = r_config_get_i (core->config, "asm.bits");
```
Comment (radare) [inline libr/core/cmd_anal.c]: imho we should have a different variable for this and not depend on asm.bits, because that will break on thumb for example

**Comment by radare** [comment]
after this change it (and rebased) it should be green

**Comment by radare** [comment]
(and ready to merge)

File: libr/anal/esil.c (hunk 1 above)
Comment (redmed666) [reply libr/anal/esil.c]: Yep indeed, that was moronic to do that :D

**Comment by redmed666** [comment]
Hum, shit, it seems that I messed up big time with my branch...

**Pull request merged**